airbyte-integrations/connectors/source-mailgun/source_mailgun/source.py

```diff
@@ -0,0 +1,177 @@
+        """
+        next_page: Optional[HttpUrl] = response.json().get('paging', {}).get('next')
+        return {"url": next_page} if next_page and self._pre_parse_response(response) else None
+
+    def path(
+            self,
+            stream_state: Mapping[str, Any] = None,
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Optional[Mapping[str, HttpUrl]] = None,
+    ) -> str:
+        # TODO: Requires improve URL creation in the CDK by using urllib.parse.urljoin
+        #  (airbyte_cdk.sources.streams.http.http._create_prepared_request)
+        return next_page_token["url"] if next_page_token else ""
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        yield from self._pre_parse_response(response)
+
+    @staticmethod
+    def _pre_parse_response(response: requests.Response) -> List:
+        return response.json()['items']
+
+
+class Domains(MailgunStream):
+    primary_key = "name"
+
+    def path(self, *args, next_page_token: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        # return super().path(*args, **kwargs) or "domains"  # TODO: Requires the CDK update (see MailgunStream.path())
+        path: str = "domains"
+        if next_page_token:
+            token: str = next_page_token['url'].rpartition('/')[2]
+            path = f"{path}/{token}"
+
+        return path
+
+
+# Basic incremental stream
+class IncrementalMailgunStream(MailgunStream, ABC):
+
+    def __init__(self, config: Mapping[str, Any], *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Messages are stored for 3 days, so it prevents occasional attempt to read from the start of the Epoch
+        default_shift = 60 * 60 * 24 * 3
+        self.start_timestamp = config.get("start_timestamp", time.time() - default_shift)
+        self.end_timestamp = config.get("end_timestamp")
+
+    def chunk_timestamps_range(self, start_timestamp: float, interval: Number = None) -> Iterable[Tuple[float]]:
+        """
+        Yield a tuple of beginning and ending timestamps of each day between the start timestamp and end timestamp.
+        """
+        interval = interval or 60 * 60 * 24  # 1 day
+        end = self.end_timestamp or time.time()
+        if start_timestamp > end:
+            yield start_timestamp, start_timestamp
+
+        while start_timestamp <= end:
+            end_timestamp = start_timestamp + interval
+            yield start_timestamp, end_timestamp
+            start_timestamp = end_timestamp
+
+
+class Events(IncrementalMailgunStream):
+    # TODO: Event Polling. See https://documentation.mailgun.com/en/latest/api-events.html#event-polling
+
+    cursor_field = "timestamp"
+
+    primary_key = "id"
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> \
```

Comment:
the format doesn't look like it's being `black`ed, can you check, please?
Continuation is not very good for readability
```suggestion
    def get_updated_state(
        self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]
    ) -> Mapping[str, Any]:
```